source/agora/node/FullNode.d

```diff
@@ -502,6 +504,21 @@ public class FullNode : API
         }
     }
 
+    /***************************************************************************
+
+        Periodically retrieve the missing enrollments and apply them to the
+        enrollment manager.
+
+    ***************************************************************************/
+
+    protected void catchupEnrollmentTask () nothrow
+    {
+        if (this.network.peers.empty())  // no clients yet (discovery)
+            return;
+
+        this.network.getEnrollments(this.enroll_man, this.utxo_set.getUTXOFinder());
```

Comment:
This looks very dangerous. We can't really use the UTXO finder in tasks as it is not re-entrant.
Meaning that the following may occur:
```
Task A is called, starts to use the UTXOFinder, and yield mid-task;
Task B is called, starts to use the UTXOFinder, which resets the list of double spend, and yield mid-task;
Task A resumes, continues using the UTXOFinder, unaware that it lost replay protection.
```

Reply:
Right, I missed it, I will use peekUTXO instead.

Reply:
Updated. It uses `peekUTXO` now.